Documentation/ApiOverview/TSFE/Index.rst

```diff
@@ -0,0 +1,216 @@
+.. include:: /Includes.rst.txt
+.. index:: TSFE; TypoScriptFrontendController
+.. _tsfe:
+
+====
+TSFE
+====
+
+What is TSFE?
+=============
+
+TSFE is short for :php:`\TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController`,
+a class which exists in the system extension EXT:frontend.
+
+As the name implies: A responsibility of TSFE
+is page rendering. It also handles reading from and writing to the page cache.
+For more details it is best to look in the source code.
+
+There are several contexts in which the term TSFE is used:
+
+1.  PHP: It is passed as request attribute
+    :ref:`frontend.controller <typo3-request-attribute-frontend-controller>`
+2.  PHP: It was and is available as global array :php:`$GLOBALS['TSFE']` in PHP.
+3.  TypoScript: TypoScript function :ref:`TSFE <t3tsref:data-type-gettext-tsfe>`
+    which can be used to access public properties in TSFE.
+4.  (deprecated since v9.5 and :doc:`removed in 10.0
+    <ext_core:Changelog/10.0/Breaking-88564-PageTSconfigSettingTSFEconstantsRemoved>`)
+    Page TSconfig: :typoscript:`TSFE.constants`.
+
+Focusing on the PHP part (as the TypoScript part is covered in the
+:ref:`TypoScript Reference: TSFE <t3tsref:data-type-gettext-tsfe>` page),
+this page gives an overview, what can still be used, what is deprecated,
+removed or discouraged and in which way the TSFE class itself may be
+interesting to developers.
+
+Accessing TSFE
+===============
+
+.. attention::
+
+    Some of the former public properties and methods have been changed to
+    protected or marked as internal. Often, accessing TSFE is no longer
+    necessary, and there are better alternatives.
+
+    Directly access :php:`$GLOBALS['TSFE']` only as a last resort. It is
+    strongly discouraged if not absolutely necessary.
+
+From the source:
+
+    When calling a frontend page, an instance of this object is available
+    as :php:`$GLOBALS['TSFE']`, even though the Core development strives to get
+    rid of this in the future.
+
+If access to the
+:php:`\TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController` instance is
+necessary, use the request attribute
+:ref:`frontend.controller <typo3-request-attribute-frontend-controller>`:
+
+..  code-block:: php
+
+    $frontendController = $request->getAttribute('frontend.controller');
+
```

Comment:
```suggestion

..  seealso::
    :ref:`getting-typo3-request-object`

```

This links to the page which describes how to get the request object in different contexts.